### DISCUSSION TITLE
How to build Concourse executable?

### DISCUSSION THREAD

**Discussion by rkxx08** [concourse/concourse#6639] [Help & Support]
Hello, as my company does not allow to run unsigned executable, I have to build Concourse executable myself.
But I couldn't find any manual entry for building `concourse` or `fly`.

How am I supposed to build the executables?

**Comment by taylorsilva** [comment]
Building fly is easy. Locally, from inside the concourse repo run `go install ./fly` and you'll end up with a `fly` binary on your `$GOPATH/bin`. Or you can use go build (`go build -o ./fly ./fly`).

You can kinda do the same with the Concourse binary (`go build -o ./concourse ./cmd/concourse/`) but you'll be missing pieces like the garden binary, runc, containered, the init binary and probably other stuff I can't think of right now. Doing `concourse web` will work but getting `concourse worker` to run will probably be a bit harder. You might be able to run it and follow the error messages to get it working though. 

This docker file lists the binaries you'd need: https://github.com/concourse/ci/blob/master/dockerfiles/dev/Dockerfile

The resource-types is also another thing. If you're building from scratch then you'll have no resource-types. You'll have to build those images yourself as well. Thankfully that's more straight-forward ([see the git resource as an example](https://github.com/concourse/git-resource#development)).

Hope that helps!
